src/internal/packager/template/template.go

```diff
@@ -35,7 +36,13 @@ func GetZarfVariableConfig(ctx context.Context) *variables.VariableConfig {
 		return interactive.PromptVariable(variable)
 	}
 
-	return variables.New("zarf", prompt, logger.From(ctx))
+	var l *slog.Logger
+	if logger.Enabled(ctx) {
+		l = logger.From(ctx)
+	} else {
+		l = slog.New(message.ZarfHandler{})
+	}
+	return variables.New("zarf", prompt, l)
```

Comment:
Nit: Usually elses are avoided in Go, we can use returns to express this section as: 

```suggestion
    // Use logger if enabled
	if logger.Enabled(ctx) {
		return variables.New("zarf", prompt, logger.From(ctx))
	}
	// Use message by default
	return variables.New("zarf", prompt, slog.New(message.ZarfHandler{}))
```